tools/mcp/dotnet/AzureSDKDevToolsMCP/Tools/SpecCommonTools.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Text;
+using System.Threading.Tasks;
+using AzureSDKDevToolsMCP.Helpers;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.Services.CircuitBreaker;
+using ModelContextProtocol.Server;
+
+namespace AzureSDKDSpecTools.Tools
+{
+    [Description("This type contains tools to run various common tasks in specs repo")]
+    [McpServerToolType]
+    public class SpecCommonTools(IGitHelper _gitHelper, ILogger<SpecCommonTools> _logger)
+    {
+        private IGitHelper gitHelper = _gitHelper;
+        private ILogger<SpecCommonTools> logger = _logger;
+
+        static readonly string GET_CHANGED_TYPESPEC_PROJECT_SCRIPT = "eng/scripts/Get-TypeSpec-Folders.ps1";
+
+        [McpServerTool, Description("This tool returns list of TypeSpec projects modified in current branch")]
+        public List<string> GetModifiedTypeSpecProjects(string repoRootPath, string targetBranch = "main")
+        {
+            var baseCommitSha = gitHelper.GetMergeBaseCommitSha(repoRootPath, targetBranch);
+            if (string.IsNullOrEmpty(baseCommitSha))
+            {
+                return [$"Failed to get merge base commit SHA for {repoRootPath}"];
+            }
+
+            var scriptPath = Path.Combine(repoRootPath, GET_CHANGED_TYPESPEC_PROJECT_SCRIPT);
+            if (!File.Exists(scriptPath))
+            {
+                return [$"[{scriptPath}] path is not present"];
+            }
+
+            logger.LogInformation($"Getting changed files in current branch with diff against commit SHA {baseCommitSha}");
+            try
+            {
+                var processInfo = new ProcessStartInfo
+                {
+                    FileName = "pwsh",
+                    Arguments = $"{scriptPath}  -BaseCommitish {baseCommitSha} -IgnoreCoreFiles",
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    RedirectStandardInput = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    WorkingDirectory = repoRootPath
+                };
+                using var process = Process.Start(processInfo) ?? throw new Exception($"Failed to start the process: git diff {baseCommitSha}  --name-only");
+                process.WaitForExit();
```

Comment:
I think we'll want some way to propogate the MCP server stop command (and maybe ctrl-c) down to this, otherwise this could be an annoying to debug hung server if the process stalls or takes a long time (e.g. networking issues, bugs).

FYI @scbedd assuming we end up having a common process runner service in the azsdk mcp, supporting cancellation from the MCP client or CLI user will be important.

Reply:
Yes, we need a common util function to run the process.